actors/builtin/shared.go

```diff
@@ -91,12 +91,21 @@ type MinerAddrs struct {
 	ControlAddrs []addr.Address
 }
 
+// stub for codegen
+// func (b ConfirmSectorProofsParams) MarshalCBOR(w io.Writer) error {
+// 	_, err := w.Write([]byte("aaaaa"))
+// 	return err
+// }
+
 // Note: we could move this alias back to the mutually-importing packages that use it, now that they
 // can instead both alias the v2 version.
-//type ConfirmSectorProofsParams struct {
-//	Sectors []abi.SectorNumber
-//}
-type ConfirmSectorProofsParams = builtin0.ConfirmSectorProofsParams
+type ConfirmSectorProofsParams struct {
+	Sectors                            []abi.SectorNumber
+	PrecomputeRewardPowerStats         bool
```

Comment:
This field can be removed.  The general rule when defining params types is that their structure should depend on the protocol and not the implementation within specs-actors.  IIUC this boolean is included because `ProveCommitAggregate` shares code with `ConfirmSectorProofsParams` and the subfunction is defined to use the param struct.

It's not bad to reuse params structs in the specs-actors code for different things but it's bad if doing so forces you to add bytes to message parameters that otherwise wouldn't be there.

Reply:
fixed